src/SearchResult.tsx

```diff
@@ -1,71 +1,115 @@
 import React, { useState } from "react";
-import Hightlighter from "react-highlight-words";
+import {
+  FaWindows,
+  FaLinux,
+  FaRegQuestionCircle,
+  FaLock,
+  FaLockOpen,
+  FaCircle,
+  FaRegCircle
+} from "react-icons/fa";
 
 interface SearchResultProps {
-  filename: string;
-  results: Array<any>;
-  openFunc: Function;
-  query: string;
+  name: string;
+  owner: string;
+  os: string;
+  isLocked: boolean;
+  isPoweredOn: boolean;
+  addMachineFunction: Function;
 }
 
 const SearchResult: React.SFC<SearchResultProps> = ({
-  filename,
-  results,
-  openFunc,
-  query
+  name,
+  owner,
+  os,
+  isLocked,
+  isPoweredOn,
+  addMachineFunction
 }) => {
-  const [isCollapsed, setIsCollapsed] = useState(false);
+  const getOsLogoComponent = () => {
+    // TODO: Change to switch case
+    if (os.toLowerCase() === "windows") {
+      return <FaWindows />;
+    } else if (os.toLowerCase() === "linux") {
+      return <FaLinux />;
+    } else {
+      return <FaRegQuestionCircle />;
+    }
+  };
+
+  const getLockLogoComponent = () => {
+    if (isLocked) {
+      return <FaLock />;
+    } else {
+      return <FaLockOpen />;
+    }
+  };
+
+  const getPowerLogoComponent = () => {
+    if (isPoweredOn) {
+      return <FaCircle style={{ color: "green" }} />;
+    } else {
+      return <FaRegCircle  style={{ color: "grey" }}/>;
+    }
+  };
+
   return (
-    <div className="deepsearch-searchresult noselect">
+    <div 
+      className="jupyter-mandalab-searchresult noselect"
+      onDoubleClick={(e) => {addMachineFunction();}}
+    >
       <div
         style={{
           overflow: "hidden"
         }}
       >
         <div
-          className="deepsearch-searchresult-filename deepsearch-highlight-hover"
-          onClick={() => {
-            setIsCollapsed(!isCollapsed);
-          }}
+          className="jupyter-mandalab-searchresult-details jupyter-mandalab-highlight-hover"
         >
           <span
-            className={isCollapsed ? "chevron right" : "chevron bottom"}
             style={{
-              margin: "0 0.5em 0 0",
-              display: "inline-block",
-              position: "relative",
-              bottom: "-0.2em"
+              flexBasis: '5%',
+              marginTop: '0.35em',
+            }}
+          >{getOsLogoComponent()}</span>
+          <span
+            style={{
+              flexBasis: '60%',
+              whiteSpace: 'nowrap',
+              overflow: 'hidden',
+              textOverflow: 'ellipsis'
+            }}
+          >{name}</span>
+          <span
```

Comment:
All these styles of name / owner etc. maybe better in a separated css file in my opinion.

Reply:
Actually when I use `flex` I like having the css with `flex-basis` in the code better, where it is more accessible